### ISSUE TITLE
backport borg check --verify-data option to 1.0

### ISSUE THREAD

**Issue by ThomasWaldmann** [borgbackup/borg#1574]
The only way to cryptographically verify integrity of all chunks (used by an archive, or by all archives) in borg 1.0 is via `borg extract --dry-run` (for 1 or all archives).

Considering the task, this is a annoyingly slow operation:
- for 1 archive it is almost like a full extract of the archive (except that no output files are written), inner duplication in the archive leads to same chunk being checked again and again.
- for N archives, it is N times a full extract, checking a lot of same chunks again and again.

The `borg check --verify-data` in master branch is much better here as it verifies every chunk / every object in the repo only once.

Also, if a chunk/object is defect, it tells its id. Via a debug command, the user could delete the defect o object. A `borg check --repair` would then replace the chunk in all affected files by an all-zero replacement chunk and mark the file as "repaired". If in a new backup the same chunk id is backed up again, the "repaired" files in the old archives can be healed.

Note: this could be used to check for potential pre-1.0.4 repo corruption.

**Comment by ThomasWaldmann** [comment]
maybe first #1575, then do this one.

also, the --first/--last PR has big changes for borg check...

**Comment by enkore** [comment]
> maybe first #1575, then do this one.

Yes

> also, the --first/--last PR has big changes for borg check, so, if we want to backport this feature without these big changes, it needs to be done first (and will produce a merge conflict in master).

Agreed. First get these parts done, merge into 1.0, backmerge, then --first/--last.

**Comment by ThomasWaldmann** [comment]
@enkore I am thinking about whether the backport is a good idea or a too big change, as it got more than initially expected (not just the verify-data impl, but also scan() new api fn + changes to LoggedIO - and without that verify-data would be rather slow). So, what's your opinion?

**Comment by RonnyPfannschmidt** [comment]
its a feature, when backporting a feature to a patch, something is wrong

**Comment by enkore** [comment]
It's grown kinda big and 1.0.8 is already a bigger release. Let's leave it.

**Comment by ThomasWaldmann** [comment]
note: closed ticket means "not for 1.0.x ever"
